test/test_ssltransport.py

```diff
@@ -37,24 +51,43 @@ def sample_request(binary=True):
     return request if binary else request.decode("utf-8")
 
 
-def validate_request(provided_request, binary=True):
+def validate_request(
+    provided_request: bytearray, binary: Union["Literal[True]", "Literal[False]"] = True
```

Comment:
There's a shorthand for a union of literals:

```suggestion
    provided_request: bytearray, binary: "Literal[False, True]" = True
```